hack/generator/pkg/codegen/pipeline_determine_resource_ownership.go

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+const resourcesPropertyName = astmodel.PropertyName("Resources")
+
+func determineResourceOwnership() PipelineStage {
+	return PipelineStage{
+		"Determine ARM resource relationships",
+		func(ctx context.Context, definitions astmodel.Types) (astmodel.Types, error) {
+			return determineOwnership(definitions)
+		},
+	}
+}
+
+func determineOwnership(definitions astmodel.Types) (astmodel.Types, error) {
+
+	updatedDefs := make(astmodel.Types)
+
+	for _, def := range definitions {
+		if resourceType, ok := def.Type().(*astmodel.ResourceType); ok {
+			specDef, err := getResourceSpecDefinition(definitions, def.Name(), resourceType)
+			if err != nil {
+				return nil, err
+			}
+
+			specType, err := resourceSpecTypeAsObject(specDef)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Couldn't extract resource %s spec type as object", def.Name())
+			}
+
+			childResourcePropertyTypeDef, err := extractChildResourcePropertyTypeDef(
+				definitions,
+				def.Name(),
+				specDef.Name(),
+				specType)
+			if err != nil {
+				return nil, err
+			}
+			if childResourcePropertyTypeDef == nil {
+				continue // This just means skip
+			}
+
+			childResourceTypeNames, err := extractChildResourceTypeNames(*childResourcePropertyTypeDef)
+			if err != nil {
+				return nil, err
+			}
+
+			err = createdUpdatedChildResourceDefinitionsWithOwner(definitions, childResourceTypeNames, def.Name(), updatedDefs)
```

Comment:
Should this be `createUpdatedChildResourceDefinitionsWithOwner`? Or just `updateChildResourceDefinitionsWithOwner`?